src/cli_utils.cc

```
@@ -78,69 +78,143 @@ Status LoadYAMLFile(const string& filename, YAML::Node &node) {
     }
 
     // Write alleles
-    yaml << YAML::Key << "alleles";
-    yaml << YAML::Value;
-    {
-        // Note: we might write out empty sites too
-        s = yaml_of_discovered_alleles(dsals, contigs, yaml);
-        if (s.bad()) return s;
+    for (auto &pr : dsals) {
+        os << "---" << endl;
+        YAML::Emitter yaml;
+        S(yaml_of_one_discovered_allele(pr.first, pr.second, contigs, yaml));
+        os << yaml.c_str() << endl;
     }
 
-    yaml << YAML::EndMap;
+    // special notation for end-of-file
+    os << "..." << endl;
 
     return Status::OK();
 }
 
-Status contigs_alleles_of_yaml(const YAML::Node& yaml,
-                               std::vector<std::pair<std::string,size_t> > &contigs,
-                               discovered_alleles &dsals) {
-    Status s;
-    if (!yaml.IsMap()) {
-        return Status::Invalid("not a map at top level");
+
+// In a YAML document built out of top-level sub-documents, get the next document.
+//
+// Notes:
+// -  Catch any exceptions, and return an empty node.
+static YAML::Node yaml_get_next_document(std::istream &is, bool skip_first_line = false) {
+    try {
+        char buf[200];
+        stringstream ss;
+        if (skip_first_line) {
+            // the document starts with a "---", skip it
+            is.getline(buf, 200);
+        }
+
+        while (is.good() && !is.eof()) {
+            is.getline(buf, 200);
+            size_t buf_len = strlen(buf);
+
+            if (buf_len == 3) {
+                // check if we reached the end of this top level document
+                string marker(buf);
+                if (marker == "---" || marker == "...")
+                    break;
+            }
+            ss.write(buf, buf_len);
+            ss.write("\n", 1);
+        }
+
+        // We have the entire document in memory. Convert to
+        // a YAML node and return.
+        return std::move(YAML::Load(ss.str()));
+    } catch (exception e) {
+        return YAML::Node();
```

Don't hide the error information like this. Is it problematic to return a `Status` and have the YAML node as an out parameter? If so, you could make the caller of this function handle the exception and convert it to a `Status`.